robot/web/web_c.go

```diff
@@ -259,18 +260,214 @@ func (svc *webService) propertiesFromStream(ctx context.Context, stream gostream
+// 			return ctx.Err()
+// 		case <-readyCh:
+// 		}
+// 		if len(modmaninterface.Managers) != 1 {
+// 			logger.Fatalf("expected len(modmaninterface.Managers): to equal 1, instead equaled %d", len(modmaninterface.Managers))
+// 		}
+// 		timeoutCtx, timeoutCancel := context.WithTimeout(context.Background(), time.Second)
+// 		defer timeoutCancel()
+
+// 		res, err := r.ResourceByName(camera.Named(stream.Name()))
+// 		if err != nil {
+// 			logger.Fatal(err.Error())
+// 		}
+
+// 		if _, err = modmaninterface.Managers[0].AddStream(timeoutCtx, res.Name().String()); err != nil {
+// 			logger.Fatal(err.Error())
+// 		}
+
+// 		// sub, err := camera.NewVideoCodecStreamSubscription(512)
+// 		// if err != nil {
+// 		// 	return err
+// 		// }
+// 		// // TODO: Denote what happens if the callback returns an error
+// 		// err = vcStream.SubscribeRTP(sub, func(pkts []*rtp.Packet) error {
+// 		// 	for _, pkt := range pkts {
+// 		// 		if err := stream.WriteRTP(pkt); err != nil {
+// 		// 			logger.Warn(err.Error())
+// 		// 		}
+// 		// 	}
+// 		// 	return nil
+// 		// })
+// 		// if err != nil {
+// 		// 	// Returning nil here will trigger reattempting
+// 		// 	return nil //nolint:nilerr
+// 		// }
+// 		// defer vcStream.Unsubscribe(sub)
+
+// 		select {
+// 		case <-ctx.Done():
+// 			return ctx.Err()
+// 		case <-readyCtx.Done():
+// 			return nil
+// 		}
+// 	}
+// 	for {
+// 		if err := streamLoop(); err != nil {
+// 			return err
+// 		}
+// 	}
+// }
+
+// func packetStream(
+// 	ctx context.Context,
+// 	vcStream camera.VideoCodecStream,
+// 	stream gostream.Stream,
+// 	logger logging.Logger,
+// ) error {
+// 	streamLoop := func() error {
+// 		readyCh, readyCtx := stream.StreamingReady()
+// 		select {
+// 		case <-ctx.Done():
+// 			return ctx.Err()
+// 		case <-readyCh:
+// 		}
+// 		sub, err := camera.NewVideoCodecStreamSubscription(512)
+// 		if err != nil {
+// 			return err
+// 		}
+// 		// TODO: Denote what happens if the callback returns an error
+// 		err = vcStream.SubscribeRTP(sub, func(pkts []*rtp.Packet) error {
+// 			for _, pkt := range pkts {
+// 				if err := stream.WriteRTP(pkt); err != nil {
+// 					logger.Warn(err.Error())
+// 				}
+// 			}
+// 			return nil
+// 		})
+// 		if err != nil {
+// 			// Returning nil here will trigger reattempting
+// 			return nil //nolint:nilerr
+// 		}
+// 		defer vcStream.Unsubscribe(sub)
+
+// 		select {
+// 		case <-ctx.Done():
+// 			return ctx.Err()
+// 		case <-readyCtx.Done():
+// 			return nil
+// 		}
+// 	}
+// 	for {
+// 		if err := streamLoop(); err != nil {
+// 			return err
+// 		}
+// 	}
+// }
+
+func (svc *webService) h264Stream(stream gostream.Stream) (camera.VideoCodecStreamSource, error) {
+	res, err := svc.r.ResourceByName(camera.Named(stream.Name()))
+	if err != nil {
+		return nil, err
+	}
+	cam, ok := res.(camera.Camera)
+	if !ok {
+		return nil, errors.Errorf("expected %s to implement camera.Camera", stream.Name())
+	}
+
+	return cam.VideoCodecStreamSource()
```

Comment:
TODO: 
1. Test behavior with existing camera module which can't suport h264 passthrough. 
2. Test behavior with camera module that uses the updated module.go which gets a peer connection but doesn't implement RTP H264 passthrough. Confirm GetImage is called.
3. Test behavior with a module that does support H264 passthrough